opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -501,6 +530,50 @@ def use_span(
             end_on_exit: Whether to end the span automatically when leaving the
                 context manager.
         """
+
+
+class DefaultTracer(Tracer):
+    """The default Tracer that is used when no Tracer implementation is available.
+
+    All operations are no-op.
+    """
+
+    # Constant used to represent the current span being used as a parent.
+    # This is the default behavior when creating spans.
+    CURRENT_SPAN = DefaultSpan(INVALID_SPAN_CONTEXT)
+
+    def get_current_span(self) -> "Span":
+        # pylint: disable=no-self-use
+        return INVALID_SPAN
+
+    def start_span(
+        self,
+        name: str,
+        parent: ParentSpan = CURRENT_SPAN,
+        kind: SpanKind = SpanKind.INTERNAL,
+        attributes: typing.Optional[types.Attributes] = None,
+        links: typing.Sequence[Link] = (),
+        start_time: typing.Optional[int] = None,
+    ) -> "Span":
+        # pylint: disable=unused-argument,no-self-use
+        return INVALID_SPAN
+
+    @contextmanager  # type: ignore
```

Comment:
What's the right solution for context manager types? I see we've been adding `# type: ignore` to these since (at least) #92, but couldn't find a discussion about this.

Reply:
This is tracked in #219. The best solution I know is still https://github.com/open-telemetry/opentelemetry-python/pull/198#discussion_r333399436.